x/pricefeed/simulation/operations/msg.go

```diff
@@ -0,0 +1,149 @@
+package operations
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+	"github.com/cosmos/cosmos-sdk/x/simulation"
+
+	"github.com/kava-labs/kava/x/pricefeed"
+	"github.com/kava-labs/kava/x/pricefeed/keeper"
+	"github.com/kava-labs/kava/x/pricefeed/types"
+
+	tmtime "github.com/tendermint/tendermint/types/time"
+)
+
+var (
+	noOpMsg     = simulation.NoOpMsg(pricefeed.ModuleName) // TODO QUESTION what is the purpose of noopmsg?
+	govDenom    = "ukava"                                  // TODO QUESTION do I need these in pricefeed?
+	stableDenom = "usdx"
+)
+
+// SimulateMsgUpdatePrices updates the prices of various assets
+// TODO - MUST include BNB, USDX, BTC as HARD REQUIREMENTS - asset prices needed by CDP sims
+func SimulateMsgUpdatePrices(authKeeper auth.AccountKeeper, keeper keeper.Keeper) simulation.Operation {
+	// get a pricefeed handler
+	handler := pricefeed.NewHandler(keeper)
+
+	// There's two error paths
+	// - return a OpMessage, but nil error - this will log a message but keep running the simulation
+	// - return an error - this will stop the simulation (I think)
+	return func(r *rand.Rand, app *baseapp.BaseApp, ctx sdk.Context, accs []simulation.Account) (
+		simulation.OperationMsg, []simulation.FutureOperation, error) {
+
+		// OVERALL LOGIC IDEA:
+		// (1) RANDOMLY PICK AN ASSET OUT OF BNB AN BTC [TODO QUESTION - PRESUMABLY USDX IS EXCLUDED AS IT IS A STABLE DENOM?]
+		// (2) GET THE CURRENT PRICE OF THAT ASSET IN USD
+		// (3) GENERATE A RANDOM NUMBER IN THE RANGE 0.8-1.2 (UNIFORM DISTRIBUTION) [TODO POTENTIAL UPDATE MAKE IT NORMALLY DISTRIBUTED ACROSS A WIDER RANGE?]
+		// (4) MULTIPLY THE CURRENT PRICE BY THE RANDOM NUMBER
+		// (5) POST THE NEW PRICE TO THE KEEPER
+
+		// get the address for the account
+		// TODO QUESTION do we need to check if this address exists on chain or is an oracle for the price or does it not matter
+		address := sdk.AccAddress("kava1ffv7nhd3z6sych2qpqkk03ec6hzkmufy0r2s4c")
+
+		// pick a random asset out of BNB and BTC
+		assetCode, err := pickRandomAsset(r)
+		if err != nil {
+			return noOpMsg, nil, fmt.Errorf("Error picking random asset")
+		}
+
+		fmt.Printf("Picked asset: %s\n", assetCode)
+
+		// TODO QUESTION - GET THE CURRENT PRICE OF THE ASSET?? HOW TO DO THIS?
```

Comment:
`keeper.GetCurrentPrice(ctx, marketID)` will get the current system price for the asset.

Market ID is format `ASSET_DENOM:usd`.

Reply:
🆗

Reply:
Is it case sensitive?

Reply:
Should be lowercase i.e. `bnb:usd`, not sure if it's case sensitive but just make sure it matches the associated market in the generated pricefeed module params.

Reply:
🆗